reducer/avg.go

```diff
@@ -9,20 +9,28 @@ import (
 )
 
 type AvgProto struct {
-	target   string
-	resolver expr.FieldExprResolver
+	target              string
+	resolver, tresolver expr.FieldExprResolver
 }
 
 func (ap *AvgProto) Target() string {
 	return ap.target
 }
 
+func (ap *AvgProto) TargetResolver() expr.FieldExprResolver {
+	return ap.tresolver
+}
+
 func (ap *AvgProto) Instantiate() Interface {
 	return &Avg{Resolver: ap.resolver}
 }
 
-func NewAvgProto(target string, field expr.FieldExprResolver) *AvgProto {
-	return &AvgProto{target, field}
+func NewAvgProto(target string, tresolver, resolver expr.FieldExprResolver) *AvgProto {
+	return &AvgProto{
+		target:    target,
+		tresolver: tresolver,
+		resolver:  resolver,
+	}
```

Comment:
We usually put constructors immediately after the type definition, not after its methods.

Reply:
This layout was here long before these changes (I guess that's why you wrote "usually" 🙂), but I'll fix it in this PR.

(Same response for all the other identical comments.)